src/main/java/de/hpi/bpt/argos/api/eventQuery/EventQueryEndpointImpl.java

```diff
@@ -0,0 +1,198 @@
+import de.hpi.bpt.argos.util.RestEndpointUtilImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import static spark.Spark.halt;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EventQueryEndpointImpl  implements EventQueryEndpoint {
+
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeEndpointImpl.class);
+    private static final RestEndpointUtil endpointUtil = RestEndpointUtilImpl.getInstance();
+    private static final JsonParser jsonParser = new JsonParser();
+
+    private static final String JSON_EVENT_QUERY_ATTRIBUTE = "EventQuery";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.post(EventQueryEndpoint.getCreateEventQueryBaseUri(), this::createEventQuery);
+        sparkService.delete(EventQueryEndpoint.getDeleteEventQueryBaseUri(), this::deleteEventQuery);
+        sparkService.put(EventQueryEndpoint.getEditEventQueryBaseUri(), this::editEventQuery);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        try {
+            JsonObject jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+            JsonObject jsonEventQuery = jsonBody.get(JSON_EVENT_QUERY_ATTRIBUTE).getAsJsonObject();
+
+            if (jsonEventQuery == null) {
+                halt(HttpStatusCodes.BAD_REQUEST, "no event query given in body");
+            }
+
+            EventQuery eventQuery = null;
+            try {
+                eventQuery = new EventQueryImpl();
+                eventQuery.setTypeId(jsonEventQuery.get("EventTypeId").getAsLong());
+                eventQuery.setDescription(jsonEventQuery.get("Description").getAsString());
+                eventQuery.setQuery(jsonEventQuery.get("Query").getAsString());
+            } catch (Exception e) {
+                halt(HttpStatusCodes.BAD_REQUEST, "failed to parse event query");
+            }
+
+            EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().registerEventQuery(eventQuery.getTypeId(), eventQuery);
+
+            if (!feedback.isSuccessful()) {
+                halt(HttpStatusCodes.ERROR, String.format("cannot register event type: %1$s", feedback.getResponseText()));
+            }
+
+            PersistenceAdapterImpl.getInstance().createArtifact(eventQuery, EventTypeEndpoint.getEventTypeQueriesUri(eventQuery.getId()));
+
+        } catch (HaltException halt) {
+            logger.info(String.format("cannot create event query: %1$s -> %2$s", halt.statusCode(), halt.body()));
+            throw halt;
+        } catch (Exception e) {
+            LoggerUtilImpl.getInstance().error(logger, "cannot parse request body to event query", e);

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "cannot parse request body to event query" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1192)